orchagent/nhgorch.cpp

```diff
@@ -73,24 +76,96 @@ void NhgOrch::doTask(Consumer& consumer)
 
                 if (fvField(i) == "mpls_nh")
                     mpls_nhs = fvValue(i);
+
+                if (fvField(i) == "nexthop_group")
+                {
+                    nhgs = fvValue(i);
+                    if (!nhgs.empty())
+                        is_recursive = true;
+                }
             }
 
-            /* Split ips and alaises strings into vectors of tokens. */
+            /* Split ips and aliases strings into vectors of tokens. */
             vector<string> ipv = tokenize(ips, ',');
             vector<string> alsv = tokenize(aliases, ',');
             vector<string> mpls_nhv = tokenize(mpls_nhs, ',');
+            vector<string> nhgv = tokenize(nhgs, ',');
 
             /* Create the next hop group key. */
             string nhg_str;
 
-            for (uint32_t i = 0; i < ipv.size(); i++)
+            /* Keeps track of any non-existing member of a recursive nexthop group */
+            bool non_existent_member = false;
+
```

Comment:
Do we need to throw a warning if ipv.size() is non-zero when is_recursive is set?

Reply:
Done